crashtracker/src/collector_windows/api.rs

```diff
@@ -0,0 +1,789 @@
+            &mut cb_actual,
+        )
+    }
+    .context("Failed to enumerate process modules")?;
+
+    unsafe { hmodules.set_len(cb_actual as usize / size_of::<HMODULE>()) };
+
+    // Iterate through the module handles and retrieve information
+    for &hmodule in hmodules.iter() {
+        let mut module_info = MODULEINFO::default();
+        if unsafe {
+            GetModuleInformation(
+                process_handle,
+                hmodule,
+                &mut module_info,
+                size_of::<MODULEINFO>() as u32,
+            )
+        }
+        .is_err()
+        {
+            continue;
+        }
+
+        let module_path = get_module_path(process_handle, hmodule);
+
+        module_infos.push(ModuleInfo {
+            base_address: module_info.lpBaseOfDll as u64,
+            size: module_info.SizeOfImage as u64,
+            path: module_path.ok(),
+            pdb_info: get_pdb_info(process_handle, module_info.lpBaseOfDll as u64).ok(),
+        });
+    }
+
+    Ok(module_infos)
+}
+
+/// Read a value from the target process memory.
+/// # Safety
+/// This function reads memory from another process and casts it to an arbitrary type.
+/// Make sure to perform proper validation on the data before using it, and don't use it for
+/// types that have references, because the references are relative to the target process.
+unsafe fn read_memory<T>(process_handle: HANDLE, address: u64) -> Result<T> {
+    let mut bytes_read = 0;
+    let mut value = MaybeUninit::<T>::uninit();
+    let size = size_of::<T>();
+
+    let result = unsafe {
+        ReadProcessMemory(
+            process_handle,
+            address as *const _,
+            value.as_mut_ptr() as *mut _,
+            size,
+            Some(&mut bytes_read),
+        )
+    };
+
+    anyhow::ensure!(
+        result.is_ok() && bytes_read == size,
+        "Failed to read memory"
+    );
+
+    Ok(value.assume_init())
+}
+
+fn read_memory_raw(process_handle: HANDLE, address: u64, size: usize) -> Result<Vec<u8>> {
+    let mut buffer = vec![0u8; size];
+    let mut bytes_read = 0;
+
+    let result = unsafe {
+        ReadProcessMemory(
+            process_handle,
+            address as *const _,
+            buffer.as_mut_ptr() as *mut _,
+            size,
+            Some(&mut bytes_read),
+        )
+    };
+
+    if result.is_err() || bytes_read != size {
+        return Err(anyhow!("Failed to read memory"));
+    }
+
+    Ok(buffer)
+}
+
+#[repr(C)]
+struct ImageNtHeadersGeneric {
+    pub signature: u32,
+    pub file_header: IMAGE_FILE_HEADER,
+    pub magic: IMAGE_OPTIONAL_HEADER_MAGIC,
+}
+
+#[repr(C)]
+#[derive(Debug)]
+struct Guid {
+    pub data1: u32,
+    pub data2: u16,
+    pub data3: u16,
+    pub data4: [u8; 8],
+}
+
+impl fmt::LowerHex for Guid {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{:08x}{:04x}{:04x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}",
```

Comment:
comment documenting why this format

Reply:
Done in https://github.com/DataDog/libdatadog/pull/892/commits/3c3a9de72faaa3431a4cef62a8c911b5bdd218e3